src/land/heightmap.rs

```diff
@@ -0,0 +1,68 @@
+use bevy::{prelude::Texture, render::texture::TextureFormat};
+
+#[derive(Debug)]
+pub enum SamplingError {
+    ReadOutOfBounds()
+}
+
+pub trait HeightmapData
+{
+    /// Get the bounds of this heightmap (x, y).
+    /// Sampled values must be in the [0, size-1] range.
+    fn size(&self) -> (u32, u32);
+
+    /// Sample a height from the heightmap. This allows reads one either side of the size, i.e. `-1` and `size().0` are valid sample positions
+    fn sample(&self, x:i32, y:i32) -> Result<f32, SamplingError>;
```

Comment:
Always have spaces post-colon for types
```suggestion
    fn sample(&self, x: i32, y: i32) -> Result<f32, SamplingError>;
```